app/src/main/java/com/example/anfcodetest/MainPresenter.kt

```diff
@@ -1,103 +1,51 @@
 package com.example.anfcodetest
 
-import android.content.Intent
-import android.text.Html
-import android.text.method.LinkMovementMethod
-import android.view.View.GONE
-import com.squareup.picasso.Picasso
-import kotlinx.android.synthetic.main.card_row.view.*
-
-
-class MainPresenter : MainContract.Presenter {
-
-    override fun cardSetup(holder: CustomViewHolder, position: Int, cardList: List<Card>) {
-        val card = cardList[position]
-
-        //Check if there is a card image for the current card and remove the view if not
-        if (card.backgroundImage.isNullOrBlank()) {
-            holder.view.cardImage.visibility = GONE
```

Comment:
If we move this method to the ViewHolder, we can also change how much we have to type to access properties. Instead of saying `holder.view.cardImage...` we could just write: `view.cardImage`!